biz.aQute.junit/src/aQute/junit/ConsoleReporter.java

```diff
@@ -0,0 +1,53 @@
+package aQute.junit;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+import aQute.lib.exceptions.Exceptions;
+import junit.framework.AssertionFailedError;
+import junit.framework.Test;
+import junit.framework.TestListener;
+
+public class ConsoleReporter implements TestListener {
```

Comment:
That does not seem a meaningful answer. Both are TestListeners, both are called during test execution. Why have 2 listeners, when one will do? Btw, the output of your code is captured by the Tee streams. Not sure if you intended that.